spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -121,21 +138,34 @@ def __init__(
         self.__data_types = data_types
         self.__n_neurons = n_neurons
         self.__bitfield_variables = bitfield_variables
+
         self.__per_timestep_variables = per_timestep_variables
         self.__per_timestep_datatypes = per_timestep_datatypes
         self.__per_timestep_recording = set()
+
+        self.__events_per_core_variables = events_per_core_variables
+        self.__events_per_core_datatypes = events_per_core_datatypes
+        self.__events_per_core_recording = set()
+        self.__max_rewires_per_ts = 0
```

Comment:
To support events properly, this should probably become __max_events_per_ts and be a dictionary for each event variable.